### PR TITLE
7zip: Downgrade to version 23.01 (official version)

### PR DESCRIPTION
**Pull request by alexchapar**
## What type of Apps related PR is this? (check all applicable)
- [ ] ✨ Add Manifest
- [x] ♻️ Refactor
- [x] 🐛 Bug Fix
- [ ] 👷 Optimization
- [ ] 🚩 Other
## Description
- Change the 7zip installation process since some enterprise are limiting or blocking some msi installation 😒
- Also Fix a little issue with the checkvers which included the beta version.
## Related Tickets & Documents
None, but someone should have created an issue for that  🤣
## [optional] What gif/emoji best describes this PR or how it makes you feel?
👔💩🐛
### Add/Modify a description in the manifest?
- [x] 👍 yes
- [ ] 🙋 yes, but I need help
- [ ] 🙅 no, because they aren't needed
### Check URLs?
- [x] 👍 yes
- [ ] 🙋 yes, but I need help
- [ ] 🙅 no, because they aren't needed
### Update to the latest version?
- [x] 👍 yes 
- [ ] 🙋 yes, but I need help
- [ ] 🙅 no, because they aren't needed
### Update Hashes?
- [x] 👍 yes
- [ ] 🙋 yes, but I need help
- [ ] 🙅 no, because they aren't needed
### Format the json manifest?
- [x] 👍 yes
- [ ] 🙋 yes, but I need help
- [ ] 🙅 no, because they aren't needed
### Notes in the manifest?
- [ ] 👍 yes
- [ ] 🙋 yes, but I need help
- [x] 🙅 no, because they aren't needed
### Application Dependencies?
- [ ] 👍 yes
- [ ] 🙋 yes, but I need help
- [x] 🙅 no, because they aren't needed
### Application Suggestions?
- [ ] 👍 yes
- [ ] 🙋 yes, but I need help
- [x] 🙅 no, because they aren't needed
### Pre-installation steps to perform?
- [x] 👍 yes
- [ ] 🙋 yes, but I need help
- [ ] 🙅 no, because they aren't needed
### Post-installation steps to perform?
- [x] 👍 yes
- [ ] 🙋 yes, but I need help
- [ ] 🙅 no, because they aren't needed
### File persistence to perform?
- [x] 👍 yes
- [ ] 🙋 yes, but I need help
- [ ] 🙅 no, because they aren't needed

### CODE REVIEW

**Comment by BotServiceAutomation** [comment]
All changes look good.

Wait for review from human collaborators.
### 7zip

- [x] Description
- [x] License
- [x] Hashes
- [x] Checkver
- [x] Autoupdate

**Comment by Bibz87** [comment]
PR Title should use "Downgrade" instead of "Update" :)

File: bucket/7zip.json
Comment (Bibz87) [inline bucket/7zip.json:51]: Have you tried using `$fname`?
```suggestion
        "Get-ChildItem -Path $dir -Filter $fname | Select-Object -First 1 | ForEach-Object {",
```

File: bucket/7zip.json
Comment (Bibz87) [inline bucket/7zip.json:57]:
```suggestion
        "Write-Host \"Modifying the registry files...\"",
```

**Review by Bibz87 — CHANGES_REQUESTED** [review]

File: bucket/7zip.json
Comment (alexchapar) [reply bucket/7zip.json:51]: yep i tried, but the way $fname works, it uses the last one in the list of url... which would be a reg file ... that's why I used that work around... That said, since now any exe name is the same (like 7z2301.exe) I will instead try to simplify and use the real name of the exe and remove this "search" and "foreach"...

Comment (alexchapar) [reply bucket/7zip.json]: I change the filename to use 7z.$version.exe instead and using it in the pre_install script to simplify it.

**Review by Bibz87 — APPROVED** [review]

**Pull request merged**
